pyne/endl.py

```diff
@@ -0,0 +1,341 @@
+        y1 = xs[:-1]
+        y2 = xs[1:]
+        A = - np.log(y2/y1)/np.log(x1/x2)
+        B = - (np.log(y1)*np.log(x2) - np.log(y2)*np.log(x1))/np.log(x1/x2)
+        return np.nansum(np.e**B / (A+1) * (x2**(A+1) - x1**(A+1))/de_int)
+
+    def get_rx(self, nuc, p_in, rdesc, rprop, x1=None, p_out=None):
+        """get_rx(nuc, p_in, rdesc, rprop, x1=None, p_out=None)
+        Grab the data for one reaction type.
+
+        Parameters
+        ----------
+        nuc : int
+            id form of nucleus to read from.
+        p_in : int
+            ENDL incident particle designator
+        rdesc : int
+            ENDL reaction descriptor
+        rprop : int
+            ENDL reaction property
+        x1 : int or None
+            ENDL atomic subshell indicator (if applicable)
+        yo : int or None
+            ENDL outgoing particle designator (if applicable)
+
+        Returns
+        -------
+        data : NumPy array
+            Contains the reaction data in a n-by-m-dimensional array. The
+            values of n and m depend on the reaction property rprop.
+        """
+        nuc = nucname.id(nuc)
+        if nuc in self.structure:
+            return self._read_nuc_pin_rdesc_rprop(nuc, p_in,
+                                                  rdesc, rprop,
+                                                  x1, p_out)
+        else:
+            raise ValueError('Nucleus {} does not exist.'.format(nuc))
+
+    def _read_nuc_pin_rdesc_rprop(self, nuc, p_in, rdesc, rprop, x1, yo):
+        """Load in the data from one reaction into self.structure.
+
+        Parameters
+        ----------
+        nuc : int
+            id of nuclide.
+        p_in : int
+            ENDL incident particle designator
+        rdesc : int
+            ENDL reaction descriptor
+        rprop : int
+            ENDL reaction property
+        x1 : int or None
+            ENDL atomic subshell indicator (if applicable)
+        yo : int or None
+            ENDL outgoing particle designator (if applicable)
+
+        Returns
+        -------
+        array, float64
+            float64 NumPy array containing the reaction data. The shape of the
+            array depends on the ENDL reaction property.
+        """
+        opened_here = False
+
+        try:
+            pdp_dict = self.structure[nuc]['pin_rdesc_rprop']
+        except KeyError as e:
+            msg = 'Particle {0}/reaction descriptor {1}/reaction property {2}'\
+                    ' not found.'.format(p_in, rdesc, rprop)
+            e.args = (msg,)
+            raise e
+
+        data_tuples = pdp_dict[p_in, rdesc, rprop]['data_tuples']
+
+        # Select the first data_tuple matching x1 and yo, if they are provided
+        # (i.e. not None)
+        def match_x1_yo_to_data_tuple(x, y, d):
+            return (x is None or d.x1 == x) and (y is None or d.yo == y)
+        try:
+            index, data_tuple = next((i, d) for i, d in enumerate(data_tuples)
```

Comment:
No, I'm looking for the first data tuple that matches `x1`, `yo` or both, if they are not `None`. The `enumerate` is useless, however. I used it for the caching but then forgot to remove it.